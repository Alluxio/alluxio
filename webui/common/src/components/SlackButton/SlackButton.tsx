/*
 * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
 * (the "License"). You may not use this work except in compliance with the License, which is
 * available at www.apache.org/licenses/LICENSE-2.0
 *
 * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied, as more fully set forth in the License.
 *
 * See the NOTICE file distributed with this work for information regarding copyright ownership.
 */

import React from 'react';
import { Button, PopoverBody, UncontrolledPopover } from 'reactstrap';
import slackLogo from '../../images/icon-slack.svg';
import './SlackButton.css';

const SLACK_ID = 'SlackChannel';
const SLACK_URL = 'https://www.alluxio.io/slack';

export class SlackButton extends React.PureComponent {
  public render(): JSX.Element {
    return (
      <>
        <div id={SLACK_ID} className="absolute-bottom-right">
          <Button
            id={SLACK_ID}
            className="button-icon"
            color="info"
            onClick={(): void => {
              window.open(SLACK_URL, '_blank');
            }}
          >
            <img src={slackLogo} />
          </Button>
        </div>
        <UncontrolledPopover placement="auto" target={SLACK_ID} trigger="hover">
          <PopoverBody>
            <span>
              Ask a question on{' '}
              <a href={SLACK_URL} rel="noopener noreferrer" target="_blank">
                Slack
              </a>
              !
            </span>
          </PopoverBody>
        </UncontrolledPopover>
      </>
    );
  }
}
