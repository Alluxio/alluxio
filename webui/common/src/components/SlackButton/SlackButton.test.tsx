/*
 * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
 * (the "License"). You may not use this work except in compliance with the License, which is
 * available at www.apache.org/licenses/LICENSE-2.0
 *
 * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied, as more fully set forth in the License.
 *
 * See the NOTICE file distributed with this work for information regarding copyright ownership.
 */

import { configure, shallow, ShallowWrapper } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';

import { SlackButton } from './SlackButton';

configure({ adapter: new Adapter() });

describe('SlackButton', () => {
  describe('Shallow component', () => {
    let shallowWrapper: ShallowWrapper;

    beforeAll(() => {
      shallowWrapper = shallow(<SlackButton />);
    });

    it('Renders without crashing', () => {
      expect(shallowWrapper.length).toEqual(1);
    });

    it('Matches snapshot', () => {
      expect(shallowWrapper).toMatchSnapshot();
    });
  });
});
