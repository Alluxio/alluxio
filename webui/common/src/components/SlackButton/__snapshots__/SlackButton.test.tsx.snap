// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SlackButton Shallow component Matches snapshot 1`] = `
<Fragment>
  <div
    className="absolute-bottom-right"
    id="SlackChannel"
  >
    <Button
      className="button-icon"
      color="info"
      id="SlackChannel"
      onClick={[Function]}
      tag="button"
    >
      <img
        src="icon-slack.svg"
      />
    </Button>
  </div>
  <UncontrolledPopover
    placement="auto"
    target="SlackChannel"
    trigger="hover"
  >
    <PopoverBody
      tag="div"
    >
      <span>
        Ask a question on
         
        <a
          href="https://www.alluxio.io/slack"
          rel="noopener noreferrer"
          target="_blank"
        >
          Slack
        </a>
        !
      </span>
    </PopoverBody>
  </UncontrolledPopover>
</Fragment>
`;
